### PR TITLE
Chat thread uses new pagination API

### PR DESCRIPTION
**Pull request by chrisnojima**
We use the new API exposed by core to not have to send paginationNext/prev anymore
Instead we find the first / last valid messageID we've seen and use that as the pagination 'pivot'

Additionally we pass an explicit number to load more in the sync+append case. We were sending 9999 originally which had a side effect of giving you older values which we'd then just throw away as dupes. This is now calculated (it was before) and now it's plumbed through to the load action

Found an unrelated bug where the conversation list would blank out due to props.validated being false which only was valid if the state was 'unboxed' when reUnboxing is also a valid state to keep itself shown (as in after a sync+append)

@keybase/react-hackers 
cc: @mmaxim

### CODE REVIEW

**Comment by mmaxim** [comment]
I'm going to try this out now.

File: shared/actions/chat/index.js [hunk 1]
```diff
@@ -378,6 +358,7 @@ const getThreadNonblockSagaMap = (yourName, yourDeviceName, conversationIDKey, a
 
 function* _loadMoreMessages(action: Constants.LoadMoreMessages): SagaGenerator<any, any> {
   const conversationIDKey = action.payload.conversationIDKey
+  const recent = action.payload.wantNewer === true
```
Comment (mmaxim) [inline shared/actions/chat/index.js]: I'm interested why you need `=== true` here, but I guess it is safer.

**Comment by chrisnojima** [comment]
its all good. api stuff is hard and we all thought it'd be simpler than it was

File: shared/actions/chat/index.js [hunk 2]
```diff
@@ -1274,29 +1245,30 @@ function* _inboxSynced(action: Constants.InboxSynced): SagaGenerator<any, any> {
...
-        )
-        yield all([
-          put(Creators.clearMessages(selectedConversation)),
-          yield put(Creators.loadMoreMessages(selectedConversation, false)),
-        ])
-        return
+      if (lastMessage && lastMessage.rawMessageID && inbox && inbox.maxMsgID) {
+        numberOverride = inbox.maxMsgID - lastMessage.rawMessageID
+
+        if (numberOverride > tooManyMessagesToJustAppendOnStale) {
```
Comment (mmaxim) [inline shared/actions/chat/index.js]: Also maybe this should just be the same constant as default page size?

**Pull request merged**